src/ProductConstructionService/ProductConstructionService.DependencyFlow/PullRequestUpdater.cs

```diff
@@ -905,6 +928,46 @@ await CreateCodeFlowPullRequestAsync(
         {
             await UpdateAssetsAndSources(update, pr);
         }
+        catch (ConflictInPrBranchException conflictException)
+        {
+            // The PR we're trying to update has a conflict with the source repo. We will mark it as blocked, not allowing any updates from this
+            // subscription till it's merged, or the conflict resolved. We'll set a reminder to check on it.
+            StringBuilder sb = new();
+            var commitUri = update.SourceRepo.Contains("github.com")
+                ? $"{update.SourceRepo}/tree/{update.SourceSha}"
+                : $"{update.SourceRepo}?version=GC{update.SourceSha}";
+            sb.AppendLine($"There was a conflict in the PR branch when flowing source from {commitUri}");
+            sb.AppendLine("Conflicting files:");
+            foreach (var file in conflictException.FilesInConflict)
+            {
+                sb.AppendLine($" - {file}");
+            }
+            sb.AppendLine();
+            sb.AppendLine("Updates from this subscription will be blocked until the conflict is resolved, or the PR is merged");
+
+            (var targetRepository, _) = await GetTargetAsync();
+            var remote = await _remoteFactory.CreateRemoteAsync(targetRepository);
+
+            await remote.CommentPullRequestAsync(pr.Url, sb.ToString());
+            // If the headBranch gets updated, we will retry to update it with previously conflicting changes. If these changes still cause a conflict, we should update the
+            // InProgressPullRequest with the latest commit from the remote branch
+            var remoteCommit = await remote.GetLatestCommitAsync(targetRepository, pr.HeadBranch);
+
+            await _pullRequestCheckReminders.SetReminderAsync(
```

Comment:
I think you should use `SetPullRequestCheckReminder` instead

Reply:
It can also update the `InProgressPullRequest` state for you (with the conflict)